maro/rl/algorithms/dqn.py

```diff
@@ -19,38 +18,31 @@ class DQNConfig:
         reward_discount (float): Reward decay as defined in standard RL terminology.
         target_update_freq (int): Number of training rounds between target model updates.
         train_epochs (int): Number of training epochs per call to ``update()``. Defaults to 1.
-        gradient_iters (int): Number of gradient steps for each mini-batch. Defaults to 1.
         soft_update_coefficient (float): Soft update coefficient, e.g.,
             target_model = (soft_update_coefficient) * eval_model + (1-soft_update_coefficient) * target_model.
             Defaults to 1.0.
         double (bool): If True, the next Q values will be computed according to the double DQN algorithm,
             i.e., q_next = Q_target(s, argmax(Q_eval(s, a))). Otherwise, q_next = max(Q_target(s, a)).
             See https://arxiv.org/pdf/1509.06461.pdf for details. Defaults to False.
-        loss_cls: A string indicating a loss class provided by torch.nn or a custom loss class. If it is a string,
-            it must be a key in ``TORCH_LOSS``. Defaults to "mse".
     """
     __slots__ = [
         "reward_discount", "target_update_freq", "train_epochs", "gradient_iters", "soft_update_coefficient",
-        "double", "loss_func"
```

Comment:
why remove `loss_func`?

Reply:
The original DQN uses MSE loss (see psuedo code on Page 7 of the paper), while the prioritized experience replay version uses the vanilla TD error (see lines 11 & 13 on Page 5 of the paper) due to the need for annealing with importance sampling weights. I believe these choices are based on some mathematical guarantees (although I am not 100% sure). It is very unlikely that one would want to use a loss function other than these two.

Reply:
Will it be better if we still enable this parameter and use MSE as the default loss func?

Reply:
Also, I remember that besides the proportional prioritized replay, there is another rank-based one. How about mentioning this in the docstring of this class?

Reply:
> Will it be better if we still enable this parameter and use MSE as the default loss func?

We could. My concern is that if prioritized experience replay is used, the "loss_func" parameter would be useless.

Reply:
> Also, I remember that besides the proportional prioritized replay, there is another rank-based one. How about mentioning this in the docstring of this class?

OK. I haven't quite looked at the implementation of the rank-based algorithm, though.